src/nspawn/nspawn-mount.c

```diff
@@ -598,13 +598,17 @@ int mount_all(const char *dest,
                   MOUNT_FATAL }, /* If /etc/os-release doesn't exist use the version in /usr/lib as fallback */
                 { NULL,                     "/run/host/os-release",         NULL,    NULL,                             MS_BIND|MS_RDONLY|MS_NOSUID|MS_NOEXEC|MS_NODEV|MS_REMOUNT,
                   MOUNT_FATAL },
+                { NULL,                     "/run/host/os-release",         NULL,    NULL,                             MS_PRIVATE,
+                  MOUNT_FATAL },  /* Turn off propagation (we only want that for the mount propagation tunnel dir) */
                 { NULL,                     "/run/host",                    NULL,    NULL,                             MS_BIND|MS_RDONLY|MS_NOSUID|MS_NOEXEC|MS_NODEV|MS_REMOUNT,
                   MOUNT_FATAL|MOUNT_IN_USERNS },
 #if HAVE_SELINUX
                 { "/sys/fs/selinux",        "/sys/fs/selinux",              NULL,    NULL,                             MS_BIND,
                   MOUNT_MKDIR },  /* Bind mount first (mkdir/chown the mount point in case /sys/ is mounted as minimal skeleton tmpfs) */
                 { NULL,                     "/sys/fs/selinux",              NULL,    NULL,                             MS_BIND|MS_RDONLY|MS_NOSUID|MS_NOEXEC|MS_NODEV|MS_REMOUNT,
                   0 },            /* Then, make it r/o (don't mkdir/chown the mount point here, the previous entry already did that) */
+                { NULL,                     "/sys/fs/selinux",              NULL,    NULL,                             MS_PRIVATE,
```

Comment:
If I do
```
debootstrap unstable /var/lib/machines/deb-unstable

ubuntu@jam1:~$ sudo systemd-nspawn -D /var/lib/machines/deb-unstable/
Spawning container deb-unstable on /var/lib/machines/deb-unstable.
Press Ctrl-] three times within 1s to kill container.
Failed to mount n/a (type n/a) on /var/lib/machines/deb-unstable/sys/fs/selinux (MS_PRIVATE ""): No such file or directory
```

Reply:
yes, `MOUNT_FATAL` needs to be removed here, as it won't exist on systems without selinux

Reply:
I think this generally fails as soon as the distro doesn't support selinux.